src/aal/aal_mips.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#define SNMALLOC_VA_BITS_64
+
+namespace snmalloc
+{
+  /**
+   * MIPS architecture layer, phrased as generically as possible.  Specific
+   * MIPS implementations may need to adjust some of these.
+   */
+  class AAL_MIPS
+  {
+  public:
+    static constexpr uint64_t aal_features = IntegerPointers;
+
+    static constexpr size_t smallest_page_size = 0x1000;
+
+    static constexpr int aal_name = MIPS;
+
+    static void inline pause()
+    {
+      ;
+      /*
+       * XXX the PAUSE instruction could be correct or not, depending on the
```

Comment:
I don't see a `PAUSE` instruction in the MIPS64 ISA reference.  What is this referring to?

Reply:
Terribly belatedly, MIPS64 Volume II-A, Revision 6.05 (https://s3-eu-west-1.amazonaws.com/downloads-mips/documents/MIPS_Architecture_MIPS64_InstructionSet_%20AFP_P_MD00087_06.05.pdf) defines a PAUSE instruction on page 374.  It snoops on the LL bit, which is why it may or may not be the right thing to do.